### PR TITLE
Docker Machine!

### LINKED ISSUE #418: Consider using docker-machine instead of boot2docker
Since boot2docker is deprecated (https://github.com/boot2docker/boot2docker-cli#pending-deprecation), consider supporting docker-machine instead.

### PR DESCRIPTION
**Pull request by thieman**
WIP, cc @jsingle. Closes #418.

The plan is to get 0.5.0 out before merging any of this, since I don't want to mix two huge features (NFS and Docker Machine) into the same release.

### CODE REVIEW

File: dusty/systems/docker/__init__.py [hunk 1]
```diff
@@ -33,11 +34,12 @@ def _get_set_envs():
 def get_docker_env():
     env = _get_set_envs()
     if len(env.keys()) < 3:
-        output = check_output_demoted(['boot2docker', 'shellinit'], redirect_stderr=True)
+        output = check_output_demoted(['docker-machine', 'env', constants.VM_MACHINE_NAME], redirect_stderr=True)
```
Comment (jsingle) [inline dusty/systems/docker/__init__.py]: We still need to do this shellinit equivalent?

Comment (thieman) [inline dusty/systems/docker/__init__.py]: Yes, still very necessary.

File: dusty/systems/virtualbox/__init__.py [hunk 2]
```diff
@@ -13,133 +13,69 @@
 from ...subprocess import check_and_log_output_and_error_demoted, check_output_demoted, check_call_demoted, call_demoted
 from ...log import log_to_client
 
+def _run_command_on_vm(command_list):
+    ssh_command = ['docker-machine', 'ssh', constants.VM_MACHINE_NAME]
+    ssh_command.extend([command_list])
```
Comment (jsingle) [inline dusty/systems/virtualbox/__init__.py]: Interesting way to construct the list

Comment (thieman) [inline dusty/systems/virtualbox/__init__.py]: Jordan tell us how you really feel

**Comment by thieman** [comment]
I think VBox host-only interface problems are breaking my integration tests. Not sure how to fix this other than blowing away the existing b2d VM on the test box, so will just test locally until this is ready to be merged.

File: docs/installation.md [hunk 3]
```diff
@@ -1,24 +1,24 @@
 # Installing Dusty
 
 ## Requirements
+
 You must have the following installed in order to run Dusty:
 
- * [Virtualbox](https://www.virtualbox.org/wiki/VirtualBox)
- * [boot2docker](http://boot2docker.io/)
+ * [VirtualBox](https://www.virtualbox.org/wiki/VirtualBox)
+ * [Docker Machine](https://docs.docker.com/machine/)
```
Comment (jsingle) [inline docs/installation.md]: I thought this was just going to turn into docker toolbox or something?

File: docs/why-dusty.md [hunk 4]
```diff
@@ -40,14 +40,14 @@ others. Define each container once, then tell Dusty what you want to run. It'll
 ## OS X Support
 
 Dusty delivers the power and flexibility of Docker to OS X. Here are the biggest improvements we've made
-to running vanilla `boot2docker` on your Mac:
+to running vanilla Docker Machine on your Mac:
 
 * **Port Forwarding**: Dusty uses nginx to rig up host name and port forwarding all the way from your Mac
 into a running container. You define the host name, host port, and container port in the container's spec,
 then Dusty does the rest at runtime.
 * **rsync**: Dusty uses [rsync](https://rsync.samba.org/) for all file transfer operations between your Mac
```
Comment (jsingle) [inline docs/why-dusty.md]: lol rsync

File: docs/installation.md (hunk 3 above)
Comment (thieman) [inline docs/installation.md]: I thought splitting it out this way made sense for people who don't want to use Toolbox

File: docs/why-dusty.md (hunk 4 above)
Comment (thieman) [inline docs/why-dusty.md]: Yeaaaaaahhhhhh should clean that up a bit

File: dusty/systems/virtualbox/__init__.py [hunk 5]
```diff
@@ -13,133 +13,76 @@
...
-        done
-        sleep 1
-    }
-    wait4eth1""")
-    if call_demoted(['boot2docker', 'ssh', 'test -f /var/lib/boot2docker/profile']) != 0:
-        log_to_client('Applying boot2docker 1.7 cert hack')
-        check_and_log_output_and_error_demoted(['boot2docker', 'ssh',
-                                                'sudo sh -c \'printf \"{}\" > /var/lib/boot2docker/profile\''.format(hack)])
-        check_and_log_output_and_error_demoted(['boot2docker', 'ssh', 'sudo /etc/init.d/docker restart'])
-
```
Comment (jsingle) [inline dusty/systems/virtualbox/__init__.py]: Wow this module is so much nicer now

**Comment by jsingle** [comment]
So dusty up works and everything?

File: dusty/systems/virtualbox/__init__.py (hunk 5 above)
Comment (thieman) [inline dusty/systems/virtualbox/__init__.py]: Praise Sarin

**Comment by thieman** [comment]
Seems to, but `restart` and shit seem to be fucked up on my computer right now, I need to figure out how to downgrade Python

**Pull request merged**